### PR TITLE
Memory profiling

### PR DESCRIPTION
**Pull request by javitonino**
Add some infrastructure to enable profiling:
- Moves the rbtrace gem to all environments
- Adds a memory_dump function that dumps object information as well as class names (map object_id->class_name), in order to be able to extract class names from the object dump
- An analyzer that parses the two generated files and generates a csv file of memory usage per class

CR @juanignaciosl

### CODE REVIEW

File: config/initializers/memory_profiling.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # See http://www.be9.io/2015/09/21/memory-leak/
 # This enables memory logging in order to check memory consumption (reproducing memory leaks)
+require 'objspace'
```
Comment (juanignaciosl) [inline config/initializers/memory_profiling.rb]: Do you need this to be required, always, in production?

Comment (javitonino) [inline config/initializers/memory_profiling.rb]: Yes, to be able to generate memory dumps. It is needed for the dump_all call.

**Comment by juanignaciosl** [comment]
I've added a comment about the need of `require 'objspace'`. AFAIK rbtrace should be safe to load and be run in production, but please request devops monitoring when you deploy this (cc @santisaez, @jvillarf ), and don't merge it until they're aware.

File: config/initializers/memory_profiling.rb (hunk 1 above)
Comment (juanignaciosl) [inline config/initializers/memory_profiling.rb]: I guess that requiring that gem is harmless. Please, either confirm it actually is or require and load the module with an environment variable :_)

**Comment by javitonino** [comment]
I've moved the `require objspace` to the memory_dump function, so it is only loaded when needed. Should be safer.
Anyway, it should not be a problem, objspace is core ruby and seems to be safe in production. Loading it does not appear to have any side effects, it only runs when calling a function.

Deploying as soon as tests pass.

cc @juanignaciosl

**Comment by juanignaciosl** [comment]
Ok, better. :+1: on my side, but please don't merge or deploy until you get :+1: from system team as well. @santisaez , could you take a look, please?

**Comment by juanignaciosl** [comment]
Retest this, please.

**Comment by javitonino** [comment]
Retest this, please.

**Pull request merged**